java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -539,16 +539,16 @@ public static Optional<Map<String, Object>> getGroupFormulaValuesByNameAndGroup(
         // Load data from the legacy file if not converted yet
         File dataFile = new File(getGroupPillarDir() +
                 group.getId() + "_" + name + "." + PILLAR_FILE_EXTENSION);
```

Comment:
Nitpick, maybe move this in the `if(data.isEmpty())` block so we don't have to create dataFile object everytime group formula data are requested and are already in database.

Reply:
fixed